usr/share/rear/output/USB/Linux-i386/300_create_grub.sh

```diff
@@ -51,7 +51,15 @@ if [ ! -d "$usb_boot_dir" ] ; then
     mkdir -p $v "$usb_boot_dir" || Error "Failed to create USB boot dir '$usb_boot_dir'"
 fi
 DebugPrint "Installing GRUB2 as USB bootloader on $RAW_USB_DEVICE"
-$grub_install_binary --boot-directory=$usb_boot_dir --recheck $RAW_USB_DEVICE || Error "Failed to install GRUB2 on $RAW_USB_DEVICE"
+if is_true $USING_UEFI_BOOTLOADER ; then
+    # TODO only call grub-install if legacy boot install is requested
+    # TODO use a switch case based on the target (uname -m) and possibly other info or make this a config option?
+    # Enforce legacy BIOS installation since efi was handled in 100_create_efiboot.sh
+    # see https://github.com/rear/rear/issues/2883
+    $grub_install_binary --target=i386-pc --boot-directory=$usb_boot_dir --recheck $RAW_USB_DEVICE || Error "Failed to install GRUB2 with target=i386-pc on $RAW_USB_DEVICE"
+else
+    $grub_install_binary --boot-directory=$usb_boot_dir --recheck $RAW_USB_DEVICE || Error "Failed to install GRUB2 on $RAW_USB_DEVICE"
+fi
```

Comment:
Can't one simply invoke $grub_install_binary with `--target=i386-pc` and avoid the condition? I suppose that the intent is to install GRUB as i386-pc always (both on BIOS and EFI machines), so it should be valid to use `--target=i386-pc` always.

Reply:
I agree with this logic, let's not modify the code paths that used to work before the change, even if it may make the code a bit more comlicated. At least until we understand what happens very well (I don't - I can't comment on the use of other `i386-*` targets, no idea about them ).

Reply:
for other `i386` targets the `pc` may also work but may for example not be able to detect network interfaces for some netboot later on. I am not sure if it would work in any case out there but the ones I can think of.